frontend/src/Components/HomePage/UpcomingTrips/TrafficInfo.jsx

```diff
@@ -0,0 +1,130 @@
+import { useState, useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+import { API_BASE_URL } from '../../../config';
+
+
+function TrafficInfo({ departureAirport, departureTime, userLocation }) {
+const [airportLocation, setAirportLocation] = useState(null);
+const [trafficData, setTrafficData] = useState(null);
+const [recommendedLeaveTime, setRecommendedLeaveTime] = useState(null);
+const [error, setError] = useState(null);
+
+
+// Fetch airport location data
+const fetchAirportLocation = useCallback(async (code) => {
+    try {
+    const response = await axios.get(`${API_BASE_URL}/airport-location/${code}`);
+    if (response.data && response.data.location) {
+        setAirportLocation(response.data.location);
+    } else {
+        throw new Error('Invalid airport location data');
+    }
+    } catch (error) {
+    setError(error.response?.data?.error || 'Error fetching airport location');
+    }
+}, []);
+
+
+useEffect(() => {
+    if (departureAirport) {
+    fetchAirportLocation(departureAirport);
+    }
+}, [departureAirport, fetchAirportLocation]);
+
+
+// Calculate route and traffic information
+const calculateRoute = useCallback(async () => {
+    if (!userLocation || !airportLocation) {
+    setError('Missing location data for route calculation');
+    return;
+    }
+
+
+    try {
+    setError(null);
+    const response = await axios.post(`${API_BASE_URL}/calculate-route`, {
+        origin: userLocation,
+        destination: airportLocation,
+        departureTime: departureTime.toISOString(),
+        mode: 'driving'
+    });
+
+
+    if (response.data.error) {
+        throw new Error(response.data.error);
+    }
+
+
+    if (response.data.trafficData) {
+        setTrafficData({
+        distance: response.data.trafficData.distance,
+        duration: response.data.trafficData.duration,
+        });
+        
+        const durationInMs = parseInt(response.data.trafficData.durationValue) * 1000;
+        const bufferTime = 2 * 60 * 60 * 1000; // 2 hours buffer
```

Comment:
To avoid re-calculating this value, It would be a good idea to create const in a top level.